test/support.py

```diff
@@ -273,13 +273,15 @@ def _build(tmpdir, ext, hpy_devel, hpy_abi, compiler_verbose=0, debug=None):
     #
     dist = Distribution()
     dist.parse_config_files()
-    options = dist.get_option_dict('build_ext')
     if debug is None:
         debug = sys.flags.debug
-    options['debug'] = ('ffiplatform', debug)
-    options['force'] = ('ffiplatform', True)
-    options['build_lib'] = ('ffiplatform', tmpdir)
-    options['build_temp'] = ('ffiplatform', tmpdir)
+    options_build_ext = dist.get_option_dict('build_ext')
+    options_build_ext['debug'] = ('ffiplatform', debug)
+    options_build_ext['force'] = ('ffiplatform', True)
+    options_build_ext['build_lib'] = ('ffiplatform', tmpdir)
+    options_build_ext['build_temp'] = ('ffiplatform', tmpdir)
+    options_build_py = dist.get_option_dict('build_py')
+    options_build_py['build_lib'] = ('ffiplatform', tmpdir)
```

Comment:
So the build_ext command reads the .build_lib value from the build_py command when calling its own .write_stub method. Why this is the case I have no idea.

This was a pre-existing bug that existed because we used to overwrite the output_dir in .write_stub to be .build_lib but now we no longer do that.

Reply:
I think that this code was originally copied&pasted directly from cffi (which is why it says `ffiplatform`). I don't know the details of that it does, maybe @arigo remembers better.